lib/Form/Processor/RoleFormProcessor.php

```diff
@@ -36,24 +37,23 @@ public static function getSubscribedEvents()
 
     public function processDefaultAction(FormActionEvent $event)
     {
-        // TODO: When we have role versioning, save draft here.
-        // For now, processSaveRole takes care of saving. Follow-up: EZP-24701
-        //$this->addNotification('role.notification.draft_saved');
+        /** @var \EzSystems\RepositoryForms\Data\RoleData $roleData */
+        $roleData = $event->getData();
+        $roleDraft = $roleData->roleDraft;
+        $this->roleService->updateRoleDraft($roleDraft, $roleData);
     }
 
     public function processSaveRole(FormActionEvent $event)
     {
-        /** @var \EzSystems\RepositoryForms\Data\RoleData $roleData */
-        $roleData = $event->getData();
-        $this->roleService->updateRole($roleData->role, $roleData);
+        /** @var RoleDraft $roleDraft */
+        $roleDraft = $event->getData()->roleDraft;
+        $this->roleService->publishRoleDraft($roleDraft);
```

Comment:
@pspanja It is in a separate process...? I mean, saving (updateRoleDraft) is done in `processDefaultAction`, and publishing in `processSaveRole`. Same as for content type afaict, other than the name of the method. I could rename `processSaveRole` to `processPublishRole`...?